File: tournaments/lookAndSaySequenceNextElement/lookAndSaySequenceNextElement.cs
string lookAndSaySequenceNextElement(string element) {
    var result = new StringBuilder();
    var a = 1;

    for (int i = 1; i < element.Length; i++) {
        if (element[i] == element[i - 1]) {
            ++a;
        } else {
            result.Append(a);
            result.Append(element[i - 1]);
            a = 1;
        }
    }

    result.Append(a);
    result.Append(element[element.Length - 1]);

    return result.ToString();
}

File: tournaments/lookAndSaySequenceNextElement/lookAndSaySequenceNextElement.cpp
std::string lookAndSaySequenceNextElement(std::string inputString) {

  int consecutiveEqual = 1;
  std::string result = "";
  inputString += '#';
  for (int i = 1; i < inputString.size(); i++) {
    if (inputString[i] != inputString[i - 1]) {
      result += std::to_string(consecutiveEqual);
      result += inputString[i - 1];
      consecutiveEqual = 1;
    }
    else {
      consecutiveEqual++;
    }
  }

  return result;
}
